pkg/controller/credentialsrequest/credentialsrequest_controller.go

```diff
@@ -433,8 +433,9 @@ func (r *ReconcileCredentialsRequest) Reconcile(request reconcile.Request) (reco
 
 	isStale := cr.Generation != cr.Status.LastSyncGeneration
 	hasRecentlySynced := cr.Status.LastSyncTimestamp != nil && cr.Status.LastSyncTimestamp.Add(time.Hour*1).After(time.Now())
+	hasActiveFailureConditions := checkForFailureConditions(cr)
 
-	if !isStale && hasRecentlySynced && crSecretExists {
+	if !isStale && hasRecentlySynced && crSecretExists && !hasActiveFailureConditions && cr.Status.Provisioned {
```

Comment:
Is it technically feasible to *not* set lastSyncGen and lastSyncTime, if we did not fully reconcile the cred request without error?

I feel like the intent of these fields is to indicate the last successfully synced generation and they should not be set if we hit errors.

Reply:
From slack, we don't, normally, and we're not sure how this is possible that we got into this state.